sympy/physics/vector/point.py

```diff
@@ -483,19 +483,56 @@ def vel(self, frame):
         Examples
         ========
 
-        >>> from sympy.physics.vector import Point, ReferenceFrame
+        >>> from sympy.physics.vector import Point, ReferenceFrame, dynamicsymbols
         >>> N = ReferenceFrame('N')
         >>> p1 = Point('p1')
         >>> p1.set_vel(N, 10 * N.x)
         >>> p1.vel(N)
         10*N.x
+        >>> p = Point('p')
+        >>> q = dynamicsymbols('q')
+        >>> p.set_vel(N, 10 * N.x)
+        >>> p2 = Point('p2')
+        >>> p2.set_pos(p, q*N.x)
+        >>> p2.vel(N)
+        (Derivative(q(t), t) + 10)*N.x
 
         """
 
         _check_frame(frame)
         if not (frame in self._vel_dict):
-            raise ValueError('Velocity of point ' + self.name + ' has not been'
+            starting_node = self
+            visited = []
+            queue = [starting_node]
+            req_point = None
+            while queue: #BFS to find nearest point
+                node = queue.pop(0)
+                if node not in visited:
+                    visited.append(node)
+                    for neighbor, neighbor_pos in node._pos_dict.items():
+                        try:
+                            neighbor_pos.dt(frame) #Checks if pos vector is valid
+                        except ValueError:
+                            continue
+                        else:
+                            try :
+                                neighbor._vel_dict[frame] #Checks if point has its vel defined in req frame
+                            except KeyError:
+                                queue.append(neighbor)
+                            else:
+                                req_point = neighbor # Nearest valid point
+                                break
+                if req_point is not None:
+                        break
+            else:
+                raise ValueError('Velocity of point ' + self.name + ' has not been'
                              ' defined in ReferenceFrame ' + frame.name)
+            path = self._pdict_list(req_point, 0) #List of points in shortest path
```

Comment:
You find the nearest valid point in line 524, so why do you need to extract this path? Shouldn't, `self.pos_from(req_piont).dt(frame)` give you the component of velocity you need?

Reply:
Oh yes , I forgot about this function